testing/trino-testing/src/main/java/io/trino/testing/DistributedQueryRunner.java

```diff
@@ -133,7 +134,8 @@ private DistributedQueryRunner(
             Optional<Path> baseDataDir,
             List<SystemAccessControl> systemAccessControls,
             List<EventListener> eventListeners,
-            List<AutoCloseable> extraCloseables)
+            List<AutoCloseable> extraCloseables,
+            BiFunction<TestingTrinoServer, Session, TestingTrinoClient> clientSupplier)
```

Comment:
How about using a named `interface`? It is way easier to find its usage than finding usage of `BiFunction`. Also code is easier to read.

Reply:
sure, I can introduce one